From 665f8840df98c050adc64a9e4e64cd9e3a6c0378 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20La=C3=9F?= <bevan@bi-co.net>
Date: Thu, 13 May 2021 15:31:20 +0200
Subject: [PATCH 4/4] hackrf: update gain display when restoring settings

So far, only the sliders are updated and not the LCD type numeric
displays. Since signal emission is disabled during restore, the updates
do not automatically propagate to the numeric displays.

This commit also removes leftover definitions of __KEEP_GAIN_SETTINGS__
in qt-dab.pro since this setting is gone since qt-dab 3.5.
---
 dab-maxi/qt-dab.pro                                   | 2 --
 dab-maxi/qt-devices/hackrf-handler/hackrf-handler.cpp | 4 ++++
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/dab-maxi/qt-dab.pro b/dab-maxi/qt-dab.pro
index dd91bcc..a2a83fd 100755
--- a/dab-maxi/qt-dab.pro
+++ b/dab-maxi/qt-dab.pro
@@ -346,7 +346,6 @@ CONFIG		+= PC
 #CONFIG		+= RPI
 #DEFINES	+= SHOW_MISSING
 DEFINES		+= __DUMP_SNR__		# for experiments only
-DEFINES		+=__KEEP_GAIN_SETTINGS__
 }
 
 
@@ -430,7 +429,6 @@ CONFIG		+= faad
 
 CONFIG		+= try-epg		# do not use
 DEFINES	+= __DUMP_SNR__		# for experiments only
-DEFINES		+=__KEEP_GAIN_SETTINGS__
 }
 #	devices
 #
diff --git a/dab-maxi/qt-devices/hackrf-handler/hackrf-handler.cpp b/dab-maxi/qt-devices/hackrf-handler/hackrf-handler.cpp
index cb41ec9..4a9da5d 100755
--- a/dab-maxi/qt-devices/hackrf-handler/hackrf-handler.cpp
+++ b/dab-maxi/qt-devices/hackrf-handler/hackrf-handler.cpp
@@ -208,8 +208,12 @@ int	res;
 	         AmpEnableButton, SLOT (setChecked (bool)));
 	connect (this, SIGNAL (new_vgaValue (int)),
 		 vgaGainSlider, SLOT (setValue (int)));
+	connect (this, SIGNAL (new_vgaValue (int)),
+		 vgagainDisplay, SLOT (display (int)));
 	connect (this, SIGNAL (new_lnaValue (int)),
 	         lnaGainSlider, SLOT (setValue (int)));
+	connect (this, SIGNAL (new_lnaValue (int)),
+	         lnagainDisplay, SLOT (display (int)));
 	xmlDumper	= nullptr;
 	dumping. store (false);
 	running. store (false);
-- 
2.31.1

